lib/teleterm/clusters/cluster_kubes.go

```diff
@@ -126,3 +127,37 @@ func (c *Cluster) reissueKubeCert(ctx context.Context, kubeCluster string) error
 		}))
 	}))
 }
+
+func (c *Cluster) getKube(ctx context.Context, kubeCluster string) (types.KubeCluster, error) {
+	var kubeClusters []types.KubeCluster
+	err := AddMetadataToRetryableError(ctx, func() error {
+		proxyClient, err := c.clusterClient.ConnectToProxy(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer proxyClient.Close()
+
+		authClient, err := proxyClient.ConnectToCluster(ctx, c.clusterClient.SiteName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer authClient.Close()
+
+		kubeClusters, err = kubeutils.ListKubeClustersWithFilters(ctx, authClient, proto.ListResourcesRequest{})
```

Comment:
Shouldn't we provide a filter so that the backend doesn't have to return all kubes?

Reply:
added!